### ISSUE TITLE
Creating a universal preset / simple editor for OGL

### ISSUE THREAD

**Issue by RETEavSTUDIO** [Samsuper12/OpenWallpaper-Plasma#7]
ATTENTION, THIS ISSUE SHOULD BE TAKEN AS A WISH BOARD

Hello, I'm interested in using your project to create a simple parallax-like live wallpaper. I don't want to use Steam wallpaper engine due to heavy background load on my system. And I don't want to use Linux projects that implement this as they don't have KDE Plasma integration. I have a little experience in writing OGL shaders on the Godot Engine and I think that my skills are enough to implement a simple parallax shader.  But I've come across the fact that I understand absolutely nothing in cpp programming, so I can't even load my textures into uniforms in the fragment shader in "Blink" example.

For a simpler entry threshold, one could make a universal preset, where the vertex and fragment shaders are separated into separate files, and not built into the cpp code. Also, in the shaders, all parameters will already be bound, such as the position of the mouse, etc. (and there will be comments on which parameters are responsible for what). There will also be a short documentation describing "how to load your textures into uniforms".

The ideal solution would be to create a simple editor to load textures and write shader code, but this is probably much more complicated than creating a universal preset.

(It would also be great to be able to load an animated Gif as a texture)

**Comment by Samsuper12** [comment]
Yeah. Loading *.so/*.dll on runtime was a bad idea for this project. 

OK, let's resurrect it. 

P.S. Feel free to text me in Discord if you have ideas.
